sql/src/parser.rs

```diff
@@ -1289,4 +1344,52 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_key_partition() {
+        KeyPartitionTableCases::basic();
+        KeyPartitionTableCases::default_key_partition();
+        KeyPartitionTableCases::invalid_column_type();
+    }
+
+    struct KeyPartitionTableCases;
+
+    impl KeyPartitionTableCases {
+        fn basic() {
+            let sql = r#"CREATE TABLE `demo` (`name` string TAG, `value` double NOT NULL, `t` timestamp NOT NULL, TIMESTAMP KEY(t)) PARTITION BY KEY(name) PARTITIONS 2 ENGINE=Analytic with (enable_ttl="false")"#;
+            let stmt = Parser::parse_sql(sql).unwrap();
+            assert_eq!(stmt.len(), 1);
+            match &stmt[0] {
+                Statement::Create(v) => {
+                    if let Some(Partition::Key(p)) = &v.partition {
+                        assert!(!p.linear);
+                        assert_eq!(&p.partition_key[0], "name");
+                        assert_eq!(p.partition_num, 2);
+                    } else {
+                        panic!("failed");
+                    };
+                }
+                _ => panic!("failed"),
+            }
+        }
+
+        fn default_key_partition() {
+            let sql = r#"CREATE TABLE `demo` (`name` string TAG, `value` double NOT NULL, `t` timestamp NOT NULL, TIMESTAMP KEY(t)) PARTITION BY KEY() PARTITIONS 2 ENGINE=Analytic with (enable_ttl="false")"#;
+            let stmt = Parser::parse_sql(sql);
+            assert_eq!(
+                stmt.err().unwrap(),
+                ParserError("Expected identifier, found: )".to_string())
```

Comment:
Because the error information returned above has been modified, the unit test here also needs to be modified.
``ParserError::ParserError(format!("Fail to parse partition key, err:{}", e))``

Reply:
Understood.